src/CMakeLists.txt

```diff
@@ -201,6 +201,17 @@ if(CMAKE_COMPILER_IS_GNUCXX)
   AddCompilerFlag("-Werror=return-local-addr")
 endif()
 
+# Eigen requires overaligned buffers for maximum efficiency (e.g. on AVX512 buffers may need to
+# be aligned to 64 bytes). AliceVision currently does not support this. Fortunately this is fixed
+# in C++17. While we can't upgrade to C++17 just yet, some compilers support overaligned
+# allocation feature with a separate flag, so use it if alignment is enabled in Eigen.
+# See https://eigen.tuxfamily.org/dox/group__TopicUnalignedArrayAssert.html
+if(AV_EIGEN_MEMORY_ALIGNMENT)
+  if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 7.1)
+    AddCompilerFlag("-faligned-new")
+  endif()
```

Comment:
We need an "else" statement to generate an error if we ask for alignment, but we cannot do it, isn't it?

Reply:
The idea of this option would be to activate the option for alignment when possible.
If it's not possible, I don't see the point of allowing people to make mistakes that will end up in crashes (that are quite difficult to analyze).

Reply:
Thanks, thinking about this more, I fail to see any other potential compile option that would allow the user to enable Eigen alignment and keep it from crashing. I've extended the compiler check to consider clang and MSVC too and if nothing else is matched, we throw an error.